docs/proposed/hs_reassembly.md

```diff
@@ -0,0 +1,410 @@
+# Reassembly of incoming fragmented handshake messages
+
+# Summary
+
+At the moment, TLS handshake fragmentation is not supported. In particular
+this limits the max certificate length that Mbed TLS client can receive.
+
+This PR adds the ability to reassemble the fragmented handshake messages.
```

Comment:
If we want to integrate this design document into the library, we shouldn't reference the PR from which it originates but focus on the feature that's being documented.

```suggestion

This document describes the design of the implementation of reassembly of incoming
TLS handshake messages. The fragmentation of outgoing TLS handshake messages
is currently not supported in Mbed TLS.

Fragmentation and reassemly of DTLS handshake messages is supported in Mbed TLS,
but not documented here.
```